### PR TITLE
NON-169: Adding integration tests for updating a non-association

### PR DESCRIPTION
**Pull request by ashleygyngell**

### CODE REVIEW

File: integration_tests/mockApis/nonAssociationsApi.ts [hunk 1]
```diff
@@ -78,6 +78,20 @@ export default {
     })
   },
 
+  stubGetNonAssociationToUpdate: () => {
+    return stubFor({
+      request: {
+        method: 'GET',
```
Comment (ushkarev) [inline integration_tests/mockApis/nonAssociationsApi.ts:84]: the update method is `'PATCH'`

File: integration_tests/e2e/updatePrisonerNonAssociation.cy.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    cy.task('stubSignIn')
+    cy.task('stubAuthUser')
+    cy.task('stubNomisUserCaseloads')
+    cy.task('stubOffenderSearchGetPrisonerResult', { prisonerNumber: 'A1234BC', result: davidJones })
+    cy.task('stubListNonAssociations')
+    cy.signIn()
+    cy.visit('/prisoner/A1234BC/non-associations')
+  })
+
+  it('should allow closing a non association ', () => {
```
Comment (ushkarev) [inline integration_tests/e2e/updatePrisonerNonAssociation.cy.ts:19]: should allow ~closing~ updating a non association

File: integration_tests/mockApis/nonAssociationsApi.ts (hunk 1 above)
Comment (ashleygyngell) [reply integration_tests/mockApis/nonAssociationsApi.ts:84]: My thinking here was to 'GET' the two prisoners to _then_ perform the 'PATCH' on. There was already a stub called 'GetNonAssociation' but it was returning the wrong prisoners hence naming this one 'GetNonAssociationToUpdate'.

I have updated those stubs in a recent commit to try and make that clearer ... but I could be doing something unnecessary here, so do let me know if so! 😃

Comment (ushkarev) [reply integration_tests/mockApis/nonAssociationsApi.ts:84]: aah i misread

**Review by ushkarev — APPROVED** [review]

**Review by ushkarev — APPROVED** [review]

**Pull request merged**